course5/homework4/string.cc

```diff
@@ -0,0 +1,139 @@
+#include <iostream>
+#include <string>
+#include <stdexcept>
+#include <cstdio>
+#include <cstdlib>
+
+class Flush
+{
+public:
+        Flush(const char* end)
+        {
+            this->last = end;
+        }
+
+        const char* Get()
+        {
+            return this->last;
+        }
+
+private:
+        const char* last;
+};
+
+class Stream
+{
+public:
+
+        Stream()
+        {
+            throw std::runtime_error("No file or filename provided");
+        }
+
+        Stream(const char* filename)
+        {
+            this->file = std::fopen(filename, "w");
+            if (!this->file)
+            {
+                throw std::runtime_error("File opening failed");
+            }
+        }
+
+        Stream(std::FILE* file)
+        {
+            this->file = file;
+            if (!this->file)
+            {
+                throw std::runtime_error("File opening failed");
+            }
+        }
+
+        Stream(const Stream&)
+        {
+            throw std::runtime_error("Stream is not copyable");
```

Comment:
It is much much much much better to enforce such limitations at compile-time rather than at runtime.
Try:
```
Stream(const Stream&) = delete;
```